mesa_geo/geoagent.py

```diff
@@ -25,17 +24,16 @@ class GeoAgent(Agent, GeoBase):
     Base class for a geo model agent.
     """
 
-    def __init__(self, unique_id, model, geometry, crs):
+    def __init__(self, model, geometry, crs):
         """
         Create a new agent.
 
-        :param unique_id: Unique ID for the agent.
         :param model: The model the agent is in.
         :param geometry: A Shapely object representing the geometry of the agent.
         :param crs: The coordinate reference system of the geometry.
         """
 
-        Agent.__init__(self, unique_id, model)
+        Agent.__init__(self, model)
```

Comment:
I suggest changing this to `super().__init__(model)` to use modern python conventions.

Reply:
Did you see the double inheritance `GeoAgent(Agent, GeoBase):`? How does super() work in that case? I think that's the reason it's explicit.

Reply:
I had not seen it, but you actually want to use super because of that. Super will abide by the MRO. In this case, if I recall correctly, that means left to right, so Agent first then GeoBase. See also e.g. [this StackOverflow question](https://stackoverflow.com/questions/3277367/how-does-pythons-super-work-with-multiple-inheritance)

Reply:
reading up on this a bit more, we probably should add an explicit super call in all mesa base classes (so model, agent, etc. this is required to make super work correctly in the entire hierarchy).

Reply:
Should I also change `GeoBase.__init__` or only `Agent.__init__`?

Reply:
all `__init__` methods should have a super call for MRO to work.

Reply:
Sorry, I don't completely understand, especially with how to pass the arguments in that case.

Can you do a concrete code suggestion?

Reply:
I agree, and than probably fix is for model and agent in one go